client/state/shortcodes/schema.js

```diff
@@ -0,0 +1,19 @@
+export const shortcodesSchema = {
+	type: 'object',
+	patternProperties: {
+		'^\\d+$': {
+			type: 'object',
+			patternProperties: {
+				'^.+$': {
+					type: 'object',
+					properties: {
+						result: { type: 'string' },
+						shortcode: { type: 'string' },
```

Comment:
I don't feel strongly one way or the other, and we don't have a strong history of modifying response values, but since `shortcode` is redundant with the key of the object, we could `omit` (or conversely `pick`) this from the reducer state.

(related to https://github.com/Automattic/wp-calypso/pull/8116#discussion_r84277962)